Lab_2

vitttoria opened the pull request:
В задании class Library почему-то не выдает совпадение по id книги = 2. Совпадение происходит, если закомментировать ValueError, но индекс выдаёт = 0.

vitttoria committed:
Lab_2

В задании class Library почему-то не выдает совпадение по id книги = 2. Совпадение происходит, если закомментировать ValueError, но индекс выдаёт = 0.

EgorOb commented on Lab_2/class Library.py:
```diff
@@ -0,0 +1,58 @@
+
+class Library:
+    def __init__(self, books=None):
+        self.books = books  # атрибут - список книг
```
Лучше будет обработать None
```python
        if books is None:
            books = []
        self.books = books
```

EgorOb commented on Lab_2/class Library.py:
```diff
@@ -0,0 +1,58 @@
+            return 1
+        if self.books is not None:
+            count = len(self.books)
+            return self.books[count - 1].id + 1
```
тут можно без count
```python
return self.books[-1].id + 1
```
помним, что [-1] даёт возможность обратиться к последнему элементу

EgorOb commented on Lab_2/class Library.py:
```diff
@@ -0,0 +1,58 @@
+            count = len(self.books)
+            return self.books[count - 1].id + 1
+
+    def get_index_by_book_id(self, book_id):
```
Сложную конструкцию взяли. Так проще
```python
def get_index_by_book_id(self, book_id):
    for index, book in enumerate(self.books):
        if book.id == book_id:
            return index

    raise ValueError("Книги с запрашиваемым id не существует")
```

EgorOb reviewed (changes requested):
В целом верно, поправьте эти моменты и добавьте описание функций и зачту. Помним это полезная практика, хоть и отнимает порой много времени

vitttoria committed:
Update class Library.py

Поправила замечания.
Но по одному есть небольшая проблема:  с такой конструкцией не работает код почему-то 
        if books is None:
            books = [ ]
        self.books = books

Говорит, что такого индекса не существует, когда срабатывает функция "def get_next_book_id(self):"

EgorOb reviewed (approved):
Зачтено
